share/wake/lib/system/job.wake

```diff
@@ -86,23 +89,40 @@ export data Persistence =
 
 # A Plan describes a not-yet-executed Job
 tuple Plan =
-  export Label:        String       # The label used when showing the command during execution
-  export Command:      List String  # The command-line arguments (the first is the command to run)
-  export Visible:      List Path    # Only these files should be available to the command
-  export Environment:  List String  # KEY=VALUE environment variables fed to the command
-  export Directory:    String       # The directory in which the command should be run
-  export Stdin:        String       # The file to which standard input should be connected
-  export Stdout:       LogLevel     # How should standard output be displayed during a build
-  export Stderr:       LogLevel     # How should standard error be displayed during a build
-  export Echo:         LogLevel     # Echo the command to this stream
-  export Persistence:  Persistence  # See Persistence table above
-  export LocalOnly:    Boolean      # Must run directly in the local workspace; no output detection performed
-  export Resources:    List String  # The resources a runner must provide to the job (licenses/etc)
-  export RunnerFilter: Runner => Boolean # Reject from consideration Runners which the Plan deems inappropriate
-  export Usage:        Usage        # User-supplied usage prediction; overruled by database statistics (if any)
-  export FnInputs:     (List String => List String) # Modify the Runner's reported inputs  (files read)
-  export FnOutputs:    (List String => List String) # Modify the Runner's reported outputs (files created)
-  export IsAtty:       Boolean      # Should job run in psuedoterminal
+  # The label used when showing the command during execution
+  export Label:        String
+  # The command-line arguments (the first is the command to run)
+  export Command:      List String
+  # Only these files should be available to the command
+  export Visible:      List Path
+  # KEY=VALUE environment variables fed to the command
+  export Environment:  List String
+  # The directory in which the command should be run
+  export Directory:    String
+  # The file to which standard input should be connected
+  export Stdin:        String
+  # How should standard output be displayed during a build
+  export Stdout:       LogLevel
+  # How should standard error be displayed during a build
+  export Stderr:       LogLevel
+  # Echo the command to this stream
+  export Echo:         LogLevel
+  # See Persistence table above
+  export Persistence:  Persistence
+  # Must run directly in the local workspace; no output detection performed
+  export LocalOnly:    Boolean
+  # The resources a runner must provide to the job (licenses/etc)
+  export Resources:    List String
```

Comment:
I'd clarify that these are un-interpreted strings and that they are only meaningful to the right runners.

Reply:
@JakeSiFive @mmjconolly @ag-eitilt Should I additionally state a little blurb about how this field is discouraged or something to focus for what they're specifically meant for? My understanding is that over time and in some flows, the `Resources` field is used (abused?) as a way to get custom behavior/DSLs in plans.

Reply:
Not all usage of Resources is bad, they're just opaque strings that get passed through

I guess that's up to the authors of a `Runner` to decide, as they are the ultimate intepreters of `Resources`.

Reply:
Ah, gotcha. I may have picked up my sentiment from a singular example of `Resources` being used in an unintended way at some point and extrapolated.

[Clarified here](https://github.com/sifive/wake/pull/1206/commits/1fc38d4549433709dd823c8af2fd16751eb68308)